pallets/compliance-manager/src/benchmarking.rs

```diff
@@ -168,9 +223,74 @@ impl<T: Config> ComplianceRequirementBuilder<T> {
     }
 }
 
+fn setup_conditions_bench<T: Config + TestUtilsFn<AccountIdOf<T>>>(
+    conditions: u32,
+    claims: u32,
+    issuers: u32,
+    claim_types: u32,
+) -> Vec<Condition> {
+    let issuers = make_issuers::<T>(issuers, Some(claim_types as usize));
+    let conditions = make_conditions(conditions, Some(claims as usize), &issuers);
+    conditions
+}
+
+fn conditions_bench(conditions: Vec<Condition>) {
+    let encoded = conditions.encode();
+    let decoded = Vec::<Condition>::decode(&mut encoded.as_slice())
+        .expect("This shouldn't fail since we just encoded a `Vec<Condition>` value.");
+    if !conditions.eq(&decoded) {
+        panic!("This shouldn't fail.");
+    }
+}
+
 benchmarks! {
     where_clause { where T: TestUtilsFn<AccountIdOf<T>> }
 
+    /*
+    conditions_cost {
+        let c in 1..MAX_CONDITIONS;
+
+        let conditions = setup_conditions_bench::<T>(c, MIN_CONDITION_TYPE_CLAIMS, MIN_CONDITION_ISSUERS, MIN_CONDITION_ISSUER_CLAIM_TYPES);
+    }: {
+        conditions_bench(conditions);
+    }
+
+    condition_claims_cost {
+        let c in 1..MAX_CONDITION_TYPE_CLAIMS;
+
+        let conditions = setup_conditions_bench::<T>(MIN_CONDITIONS, c, MIN_CONDITION_ISSUERS, MIN_CONDITION_ISSUER_CLAIM_TYPES);
+    }: {
+        conditions_bench(conditions);
+    }
+
+    condition_issuers_cost {
+        let i in 1..MAX_CONDITION_ISSUERS;
+
+        let conditions = setup_conditions_bench::<T>(MIN_CONDITIONS, MIN_CONDITION_TYPE_CLAIMS, i, MIN_CONDITION_ISSUER_CLAIM_TYPES);
+    }: {
+        conditions_bench(conditions);
+    }
+
+    condition_issuer_claim_types_cost {
+        let t in 1..MAX_CONDITION_ISSUER_CLAIM_TYPES;
+
+        let conditions = setup_conditions_bench::<T>(MIN_CONDITIONS, MIN_CONDITION_TYPE_CLAIMS, MIN_CONDITION_ISSUERS, t);
+    }: {
+        conditions_bench(conditions);
+    }
+    */
```

Comment:
Cleanup?

Reply:
I wanted to show the results of trying to split the complexity parameters into multiple cost functions.

Reply:
Originally I wanted to use these cost functions, but the generated weights were wrong.  Each of the generated cost functions were added a large fixed weight, but we only want to increase the cost for large inputs.